Add vec256 test into linux CI tests

glaringlee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44912 Add vec256 test into linux CI tests**

This is to add vec256 test (introduced in #42685) into linux CI system.
The whole test will last 50 to 60 seconds.

Differential Revision: [D23772923](https://our.internmc.facebook.com/intern/diff/D23772923)

Commit:
Add vec256 test to linux CI

[ghstack-poisoned]

Commit:
Update on "[WIP] Add vec256 test to linux CI"




[ghstack-poisoned]

Commit:
Update on "[WIP] Add vec256 test to linux CI"


This is to add vec256 test into linux CI system.
The whole test will last 50 to 70 seconds. 

[ghstack-poisoned]

Commit:
Update on "Add vec256 test into linux CI tests"


This is to add vec256 test (introduced in #42685) into linux CI system.
The whole test will last 50 to 60 seconds.

Differential Revision: [D23772923](https://our.internmc.facebook.com/intern/diff/D23772923)

[ghstack-poisoned]

Commit:
Update on "Add vec256 test into linux CI tests"


This is to add vec256 test (introduced in #42685) into linux CI system.
The whole test will last 50 to 60 seconds.

Differential Revision: [D23772923](https://our.internmc.facebook.com/intern/diff/D23772923)

[ghstack-poisoned]

malfet commented on .jenkins/pytorch/test.sh:
```diff
@@ -352,6 +352,21 @@ test_cpp_extensions() {
+    vec256_tests=$(find . -maxdepth 1 -executable -name 'vec256_test*')
+    for vec256_exec in $vec256_tests
+    do
+      $vec256_exec
```
Please report test results in xml format as well (so they could be parsed by CI)
```suggestion
      $vec256_exec --gtest_output=xml:test/test-reports/vec256/$vec256_exec.xml
```

malfet commented on .jenkins/pytorch/test.sh:
```diff
@@ -352,6 +352,21 @@ test_cpp_extensions() {
+test_vec256() {
+  # This is to test vec256 instructions DEFAULT/AVX/AVX2 (platform dependent, some platforms might not support AVX/AVX2)
+  if [[ "$BUILD_ENVIRONMENT" != *asan* ]] && [[ "$BUILD_ENVIRONMENT" != *rocm* ]]; then
+    echo "Testing vec256 instructions"
```
```suggestion
    echo "Testing vec256 instructions"
     mkdir -p test/test-reports/vec256
```

glaringlee committed:
Update .jenkins/pytorch/test.sh

Co-authored-by: Nikita Shulga <nshulga@fb.com>

glaringlee committed:
Update .jenkins/pytorch/test.sh

Co-authored-by: Nikita Shulga <nshulga@fb.com>

Commit:
Update on "Add vec256 test into linux CI tests"


This is to add vec256 test (introduced in #42685) into linux CI system.
The whole test will last 50 to 60 seconds.

Differential Revision: [D23772923](https://our.internmc.facebook.com/intern/diff/D23772923)

[ghstack-poisoned]

Commit:
Update on "Add vec256 test into linux CI tests"


This is to add vec256 test (introduced in #42685) into linux CI system.
The whole test will last 50 to 60 seconds.

Differential Revision: [D23772923](https://our.internmc.facebook.com/intern/diff/D23772923)

[ghstack-poisoned]
